src/Interface/Modules/Render/ES/SRInterface.cc

```diff
@@ -834,6 +834,9 @@ namespace SCIRun {
 
       DEBUG_LOG_LINE_INFO
 
+      if(!mCore.getStaticComponent<ren::StaticVBOMan>() ||
```

Comment:
This check should be extracted into a function as well. It's good practice for refactoring anyway; don't add any more complexity than is needed (less to refactor later).

Reply:
I feel as though this one is a bit short for its own function. I did, however, notice that we cast a shared pointer to a weak pointer only to lock it right after am I mistaken in thinking this is pointless?